### DISCUSSION TITLE
Find two matrix M and N

### DISCUSSION THREAD

**Discussion by boubekeur27** [yalmip/YALMIP#1339] [General discussion] [answered]
non-singular matrices $M$ and $N$ such that :

$MEN=\begin{pmatrix}
1 & 0 & 0 & 0 \\
0 &1 & 0 & 0 \\
0 & 0& 0 & 0 \\
0 & 0 & 0 & 0 \\
\end{pmatrix}  $
$E=\begin{pmatrix}
1 & 2 & 3 & 0 \\
1 & 2 & 3 & 2 \\
1 & 2 & 3 & 1 \\
1 & 2 & 3 & 0 \\
\end{pmatrix}$

**Comment by johanlofberg** [comment, marked as answer]
Well, the equality is just a nonlinear nonconvex quadratic constraint. Hard, but doable. The problem is non-singular, as that is almost impossible to represent (all matrices are arbitrarily close to being non-singular) so you will have to formulate in a better way such as N'*N=I or something like that

**Reply by boubekeur27** [reply]
Is this the solution?
n=4; 
E=[-1 -1 -1 -2;4 3 2 5;-6 0 2 2;1 1 1 3];
op=rref([E,eye(n)]);
M=op(:,n+1:2*n)
y=op(:,1:n);
z=rref([y,eye(n)]);
N=z(:,n+1:2*n)

**Reply by johanlofberg** [reply]
I don't see what it has to do with YALMIP. If it solves your problem, well then by definition it is your solution
